src/components/Form/TextInput.tsx

```diff
@@ -92,20 +103,31 @@ class BaseTextInput extends Component<TextInputProps> {
         ) : (
           <Paragraph {...labelTextProps}>{labelText}</Paragraph>
         )}
+        {hintText && (
+          <Paragraph className={hintTextClassName} id={generatedHintTextId}>
+            {hintText}
+          </Paragraph>
+        )}
         <HtmlDiv className={statusTextContainerClassName}>
           <HtmlDiv {...inputContainerProps}>
             <HtmlInput
               id={propId}
               {...passProps}
               className={classnames(inputBaseClassName, inputClassName)}
-              type="text"
-              aria-describedby={generatedId}
+              type={type}
+              aria-describedby={[
+                generatedStatusTextId,
+                generatedHintTextId,
+              ].join(' ')}
```

Comment:
I'd keep this logic outside of the template. The classnames util might actually work for this as well, but wouldn't still use it for this use case.

Reply:
Logic is now moved from the template.
_Also fixes that non existing ids are given to aria._